src/ring_theory/mv_polynomial/homogeneous.lean

```diff
@@ -111,48 +83,47 @@ variables {σ R}
 lemma is_homogeneous_monomial (d : σ →₀ ℕ) (r : R) (n : ℕ) (hn : ∑ i in d.support, d i = n) :
   is_homogeneous (monomial d r) n :=
 begin
-  intros c hc,
-  classical,
-  rw coeff_monomial at hc,
-  split_ifs at hc with h,
-  { subst c, exact hn },
-  { contradiction }
+  simp_rw degree'_eq_weighted_degree' at hn,
+  exact is_weighted_homogeneous_monomial 1 d r hn
 end
+
 variables (σ) {R}
 
 lemma is_homogeneous_of_total_degree_zero {p : mv_polynomial σ R} (hp : p.total_degree = 0) :
   is_homogeneous p 0 :=
 begin
-  erw [total_degree, finset.sup_eq_bot_iff] at hp,
-  -- we have to do this in two steps to stop simp changing bot to zero
-  simp_rw [mem_support_iff] at hp,
-  exact hp,
+  intros m hm,
+  simp only [weighted_degree'_one],
+  apply finsum_eq_zero_of_forall_eq_zero,
+  intro x,
+  by_cases hx : x ∈ m.support,
+  { change _ = ⊥ at hp,
+    simp_rw [total_degree, finset.sup_eq_bot_iff, mem_support_iff] at hp,
+    specialize hp m hm,
+    change _ = 0 at hp,
+    simp only [finsupp.sum, finset.sum_eq_zero_iff] at hp,
+    exact hp x hx, },
+  { simp only [finsupp.mem_support_iff, not_not] at hx,
+    exact hx, },
```

Comment:
Why did this proof come out longer when all the other ones came out shorter after the refactor?